internal/stream/event_stream_test.go

```diff
@@ -0,0 +1,185 @@
+package stream_test
+
+import (
+	"context"
+	"errors"
+	"io/ioutil"
+	"testing"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/golang/mock/gomock"
+	"github.com/google/uuid"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/assisted-service/internal/stream"
+	"github.com/openshift/assisted-service/models"
+	"github.com/sirupsen/logrus"
+)
+
+var _ = Describe("Notify to event stream", func() {
+	var (
+		ctx       = context.Background()
+		writer    *stream.MockEventStreamWriter
+		logger    *logrus.Logger
+		ctrl      *gomock.Controller
+		clusterID strfmt.UUID
+	)
+
+	BeforeEach(func() {
+		logger = logrus.New()
+		logger.Out = ioutil.Discard
+		ctrl = gomock.NewController(GinkgoT())
+		writer = stream.NewMockEventStreamWriter(ctrl)
+		clusterID = strfmt.UUID(uuid.New().String())
+
+	})
+
+	When("Closing event stream", func() {
```

Comment:
I'm assuming `When` is just `Context`? I'd just stick with `Context` since that's what is overwhelmingly used in the rest of the project.

For a single test context with no significant setup, just add the context string to the example (`It`).
In this case I would create a `Describe` block for each method you're testing then `It` blocks for each test like this:

```go
Describe("Close", func() {
  It("closes the underlying writer", func() {
  })

  It("succeeds when the underlying writer is nil", func() {
  })
})

Describe("Notify", func() {
  It("doesn't attempt to write to a nil writer", func() {
  })

  It("fails for a nil event", func() {
  })
  ...
})
```

Reply:
Yes, but it reads more naturally as plain English. However I agree, it's better to conform to the project's standard to cause least surprise when reading these tests.